### ISSUE TITLE
Proposal: Better number literals

### ISSUE THREAD

**Issue by bigyihsuan** [bigyihsuan/International-Phonetic-Esoteric-Language#5]
**Proposal:** Numbers shall be notated as such:
```
[ NUMBER ]
```

`NUMBER` matches the regex `[0-9A-Fa-f]+`, so at least one digit in hexadecimal from `[0, 15]`. Any other characters will be ignored. Upon reaching the `]`, the number as hexadecimal in the brackets will be pushed to the stack.

This will allow for more concise and readable creation of larger numbers. Instead of `FF1tθ` for 272, `[110]` can be pushed instead (5 bytes vs 5 bytes, but more readable).

**Examples:**
```
A == push 0xA
10 == push 0x1, push 0x0
[10] == push 0x10
[AaB] == push 0xAAB
[ffff] == push 0xFFFF
```

**Comment by lyxal** [comment]
Another idea is to have the `[` delimit a number and read the number until a non-number character is found. Therefore, `FF1tθ...` would be `[110...`, 1 byte shorter.

**Comment by bigyihsuan** [comment]
Closed with commit d76cc49.
